ee/query-service/app/db/reader.go

```diff
@@ -0,0 +1,28 @@
+package db
+
+import (
+	"github.com/ClickHouse/clickhouse-go/v2"
+
+	"github.com/jmoiron/sqlx"
+
+	basechr "go.signoz.io/signoz/pkg/query-service/app/clickhouseReader"
+)
+
+type ClickhouseReader struct {
+	conn  clickhouse.Conn
+	appdb *sqlx.DB
+	*basechr.ClickHouseReader
+}
+
+func NewDataConnector(localDB *sqlx.DB, promConfigPath string) *ClickhouseReader {
+	ch := basechr.NewReader(localDB, promConfigPath)
```

Comment:
@mindhash should we use existing reader object from basechr rather than creating a new one?

Reply:
The reader is created here for the first time in EE binary. Not sure what you mean by existing reader object. 
The NewDataConnector is created on top of base reader, to support EE specific methods in future.